site/docs/master/plugins.md

```diff
@@ -1,110 +1,99 @@
+
+```
+# Run just the e2e plugin
+$ sonobuoy run --plugin e2e
+
+# Run your custom plugin and the systemd-logs gatherer
+$ sonobuoy run --plugin customPlugin.yaml --plugin systemd-logs
 ```
 
-#### Contract
+> Note: All of the CLI options impact the generated YAML. If you would like to edit the YAML directly or see the impact your options have on the YAML, use `sonobuoy gen <your options>`.
 
-A definition file defines a container that runs the tests. This container
-can be anything you want, but must fulfil a contract.
+## How Plugins Work
+
+A plugin consists of two parts:
+
+* the core logic which runs tests/gathers data (typically a single container)
+* a sidecar added by Sonobuoy which reports the data back to the aggregator
 
 After your container completes its work, it needs to signal to Sonobuoy that
-it's done. This is done by writing out a filename to a results file. The default
+it's done by writing out the name of the results file into a "done file". The default
 value is `/tmp/results/done`, which you can configure with the `ResultsDir` value 
 in the Sonobuoy config.
 
-Sonobuoy waits for the `done` file to be present, then transmits the indicated
-file back to the aggregator. The results file is opaque to Sonobuoy, and is
-made available in the Sonobuoy results tarball in its original form.
-
-If you need additional mounts besides the default `results` mount that Sonobuoy
-always provides, you can define them in the `extra-volumes` field.
+The Sonobuoy sidecar waits for the `done` file to be present, then transmits the indicated
+file back to the aggregator.
 
-#### Choosing which plugins to run
+![sonobuoy plugins diagram][diagram]
+[diagram]: img/plugin-contract.png
 
-All of the plugin definition files get mounted as files on the aggregator pod which runs them.
+### Writing your own plugin
 
-The aggregator loads all the plugins it finds, but a separate list controls which plugins actually get run. There is a separate `config.json` which gets mounted on the aggregator which sets the configuration options for the aggregator. It has a field `Plugins` which is an array of plugin names. The default value includes both the e2e and systemd-logs plugin:
+Use the `sonobuoy gen plugin` command to help generate the YAML for your plugin definition. Once you've saved that YAML locally, you can run your plugin via:
 
 ```
-"Plugins":[{"name":"e2e"},{"name":"systemd-logs"}]
+sonobuoy run --plugin myPlugin.yaml
 ```
 
-If you want to prevent one of those plugins from  being run, simply remove that item from the list. Likewise, if you'd like to run your own custom plugin, you need to add it to this list (in addition to adding its definition file to the plugin configmap):
+For a thorough walkthrough of how to build a custom plugin from scratch, see our [blog post][customPluginsBlog] and our [example plugins][examplePlugins].
 
-```
-"Plugins":[{"name":"custom-plugin"},{"name":"systemd-logs"}]
-```
+## Plugin Result Types
```

Comment:
I think it would be good to add a link to your results blog post at the end of this section.